package cms_8120;
option java_package = "com.hikvision.finance.fms.cms.hpp.csclient";
option java_outer_classname = "ProductProto";
option java_generic_services = true;

message ReqProduct
{
    optional int32 cmd                	    = 1;    // CMD_DT_PRODUCT_LIST，CMD_DT_PRODUCT_INFO
    optional int32 id                       = 2;    // 在数据库中的流水号
    optional int32 pId                      = 3;    // 父类型ID(ProductType的ID)
	optional int32 inputUser			    = 4;	// 理财系统建单用户Id
	optional string strCode			        = 5;	// 产品编码
}

//产品表
message Product
{
    optional int32 id                       = 1;    // 在数据库中的流水号
    optional int32 pId                      = 2;    // 父类型ID(ProductType的ID)
    optional string productType             = 3;    // 产品类型名称
    optional string strName                 = 4;    // 产品名称
	optional string strCode                 = 5;    // 产品编码
    optional string strEndDate              = 6;    // 产品到期日期
	optional int32 guaranteeType			= 7;	//保障周期类型
	optional int32 guaranteeYears			= 8;	//保障期限（默认年）
	optional int32 saveType				    = 9;	//视频保存时间类型
	optional int32 saveYears				= 10;	//到期后保存时间（默认年）
	optional string strGrade				= 11;	//发行机构
	optional string strRisklevel			= 12;	//风险等级
	optional string wordTemplate			= 13;	//话术模板
}

message ProductList
{
    optional int32 cmd                      = 1;    // CMD_DT_PRODUCT_LIST
    optional int32 result                   = 2;    // 验证结果，0-验证通过，其它值验证失败
    optional string result_info             = 3;    // 信息
    repeated Product product                = 4;    // 产品列表
}

message ProductInfo
{
	optional int32 cmd                      = 1;    // CMD_DT_PRODUCT_INFO
    optional int32 result                   = 2;    // 验证结果，0-验证通过，其它值验证失败
    optional string result_info             = 3;    // 信息
    optional Product product                = 4;    // 产品列表
}

service ProductService {
	
	rpc GetProductList(ReqProduct) returns (ProductList);	       // CMD_DT_PRODUCT_LIST(获取产品信息List)
	rpc GetProductInfo(ReqProduct) returns (ProductInfo);          // CMD_DT_PRODUCT_INFO(获取单个产品的产品详情)
	rpc GetProductWordTemplate(ReqProduct) returns (ProductInfo);  // CMD_DT_PRODUCT_WORD_TEMPLATE(获取单个产品的产品话术)
}

